src/libraries/System.Net.Http/src/System/Net/Http/Metrics/MetricsHandler.cs

```diff
@@ -209,6 +221,14 @@ private static object GetBoxedStatusCode(int statusCode)
                 : statusCode;
         }
 
+        private static string GetStatusCodeString(int statusCode)
```

Comment:
@stephentoub should we consider bumping `SmallNumberCacheLength` to 512 instead and we can delete this code?

Reply:
Would we do it in 8.0 as well? If not I would suggest to do that in a separate PR.

Reply:
Or delete it regardless; the cache size is currently 300, so all success status code are covered.

I'm not against also raising the size, but we do need to be careful with it creeping up too high.

Reply:
This method will be never called with success status codes. (I will address https://github.com/dotnet/runtime/pull/91910/files#r1323214809.)